superset/utils/decorators.py

```diff
@@ -29,3 +35,51 @@ def stats_timing(stats_key, stats_logger):
         raise e
     finally:
         stats_logger.timing(stats_key, now_as_float() - start_ts)
+
+
+def etag_cache(max_age, check_perms=bool):
+    """
+    A decorator for caching views and handling etag conditional requests.
+
+    The decorator caches the response, returning headers for etag and last
+    modified. If the client makes a request that matches, the server will
+    return a "304 Not Mofified" status.
+
+    """
+    def decorator(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            try:
+                check_perms(request)
+                # build the cache key from the function arguments and any other
+                # additional GET arguments (like `form_data`, eg).
+                key_args = list(args)
+                key_kwargs = kwargs.copy()
+                key_kwargs.update(request.args)
+                cache_key = wrapper.make_cache_key(f, *key_args, **key_kwargs)
+                response = cache.get(cache_key)
+            except Exception:  # pylint: disable=broad-except
```

Comment:
Maybe there is merit in following the pattern in Flask-Caching per [here](https://github.com/sh4nks/flask-caching/blob/master/flask_caching/__init__.py#L403) where the error is re-raised in debug mode.

Reply:
Ah, good point. I'll do that.

Reply:
Can we remove the pylint exception while we're in here? I'm not seeing that anywhere else in the codebase despite similarly structured except blocks.

Reply:
@DiggidyDave the reason you're probably not seeing it is we have [this](https://github.com/apache/incubator-superset/blob/master/superset/views/base.py#L17) defined in a number of files. It's a place holder when we enforced that new code could not have any `pylint` warnings.

Reply:
Gotcha, thanks for the info. If this is the first new code that isn't grandfathered into a whole file disable directive, should we consider either 1) making this adhere to the linter rules (ie don't disable it here) if this is a warning we care about or 2) adding a blanket exclusion to `.pylintrc` if this is a warning we never care about (which seems to be the case in practice)?

If this is something that we globally don't care about at this particular moment, it seems like a good idea to just codify and enforce that in one place in the central config.

Reply:
@DiggidyDave catching a broad exception is generally bad hence why the `pylint` rule exists, i.e.,
```
try:
     int('x')
except ValueError:
    ...
```

is preferred over,
```
try:
     int('x')
except Exception:
    ...
```
That said there are rare occasions when using the later form make sense, i.e., i) the caller may be agnostic of the underlying function being called, or ii) all exceptions need to correctly corralled. In this example I think the current logic is correct. Additionally it mimics [Flask-Caching](https://github.com/sh4nks/flask-caching/blob/master/flask_caching/__init__.py#L387).

Reply:
I understand why they are bad, and I agree this is ok here. However, linter rules that are _always_ ignored in an ad-hoc way (ie not configured globally) sometimes can raise a red flag wrt code hygiene for me, so I thought I'd bring it up here while I'm ramping up on the codebase. :-) Thanks for your responses! I think we're on the same page.